GoogleForCreators/web-stories-wp
# Add eyedropper

## Context

First pass, pretty functional, `html-to-image` fixes are not yet here so don't bother with rendering issues (generated image can be different than the actual canvas).
Working on To-do items.

## Summary
https://user-images.githubusercontent.com/21036224/122219859-22e60d00-ceb0-11eb-9fad-cbde7b912e5a.mov

## Relevant Technical Choices

- To maintain the color picker popup state/focus I hide it visually (`visibility: hidden`).
- I'm starting the image generation on icon mouse enter - should gain us 20-60ms - WDYT?

## To-do

- [x] ~~Zoom support~~ Reset zoom
- [x] Support all cases for closing the eyedropper
- [x] Performance check (alternative approach to zoom - using divs as pixels) - UPDATE: perf is good
- [x] More cross-browser testing
- [x] `html-to-image` rendering fixes (only locally now)
- [x] `html-to-image` optimizations
- [x] Better eyedropper icon?
- [ ] Nicer zoom?
- [ ] Custom cursor? (cur/png)

## User-facing changes



## Testing Instructions



### QA






This PR can be tested by following these steps:

In general, you should be able to pick any color you see (on the canvas) and give it to another element.
1. Insert shape and give it a color
2. Insert another shape and give it the same color using the eyedropper
3. Insert image and pick a color from it to one of the shapes
4. etc
5. Verify at least for Chrome / Safari and in RTL mode.

### UAT

Same^







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #262
